fem/gridfunc.cpp

```diff
@@ -416,13 +416,24 @@ void GridFunction::GetNodalValues(int i, Array<double> &nval, int vdim) const
 
    if (FElem->GetRangeType() == FiniteElement::SCALAR)
    {
-      MFEM_ASSERT(FElem->GetMapType() == FiniteElement::VALUE,
-                  "invalid FE map type");
       Vector shape(dof);
-      for (k = 0; k < n; k++)
+      if (FElem->GetMapType() == FiniteElement::VALUE)
+      {
+         for (k = 0; k < n; k++)
+         {
+            FElem->CalcShape(ElemVert->IntPoint(k), shape);
+            nval[k] = shape * ((const double *)loc_data + dof * vdim);
```

Comment:
Since we are deprecating the casts from `Vector` to `const double *` in #3194, we should use an alternative way to get the offset pointer. In #3194, I switched to the following:
```c++
            nval[k] = shape * (&loc_data[dof * vdim]);
```